controllers/progressivesync_controller.go

```diff
@@ -332,6 +332,28 @@ func (r *ProgressiveSyncReconciler) updateStatusWithRetry(ctx context.Context, p
 	return retryErr
 }
 
+func (r *ProgressiveSyncReconciler) addSyncedAtAnnotation(ctx context.Context, app argov1alpha1.Application, stageName string) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		key := client.ObjectKeyFromObject(&app)
+		latest := argov1alpha1.Application{}
+		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			return err
+		}
+		val, ok := app.Annotations[utils.ProgressiveSyncSyncedAtStageKey]
+		if !ok {
```

Comment:
I am generally not a big fan of comments, but this feels like a place where we should add context on why this is here and explain how the `omitempty` works. If I were a first-time code reader, I'd certainly question this `if` clause here.